src/pages/gen2/reference/amplifyconfiguration/index.mdx

```diff
@@ -16,3 +16,44 @@ In Amplify (Gen 2), the CLI will generate an `amplifyconfiguration.json` file wi
 You can also manually create this file for a specified Amplify app ID and branch, or an AWS CloudFormation stack name with [`amplify generate config`](/gen2/reference/cli-commands#amplify-generate-config).
 
 {/* @TODO full type/interface to reference */}
+
+## Extending Amplify configuration file
+
+The `amplifyconfiguration.json` file is not just a static artifact; it's designed to be extendable to suit the evolving needs of your application. By leveraging the `addOutput` method you can programmatically add or modify configurations.
+This is particularly useful for customizing outputs that are not directly exposed through the Amplify constructs or for dynamically adjusting your app's configuration in response to changes in your backend strategy.
+
+One common scenario where extending the configuration becomes handy is when you need to add custom outputs or overwrite existing configurations without manual file edits.
+
+### Override configuration
+
+Consider a scenario where you want to override output parameters in your `amplifyconfiguration.json` that specify an S3 bucket and its region that your application will use for storing files.
+
+```ts title="amplify/backend.ts"
+const backend = defineBackend({
+  auth, 
+  data, 
+});
+
+backend.addOutput({
+  aws_user_files_s3_bucket: "amplify-Gen2-bucket",
```

Comment:
```suggestion
  aws_user_files_s3_bucket: "my-externally-managed-bucket",
```

this example is slightly confusing since we refer to this bucket as "amplify-gen2-bucket" but are not using `defineStorage`